### PR TITLE
refactor(src/explore/comp/controls/metricControl): migrate Enzyme test to RTL syntax

### PR DESCRIPTION
**Pull request by hainenber**
### SUMMARY
refactor(src/explore/comp/controls/metricControl): migrate Enzyme test to RTL syntax

### TESTING INSTRUCTIONS

CI checks are passed

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx [hunk 1]
```diff
@@ -16,38 +16,38 @@
...
-
-  function setup(props) {
-    return shallow(<MetricDefinitionOption store={store} {...props} />).dive();
-  }
-
-  it('renders a StyledMetricOption given a saved metric', () => {
-    const wrapper = setup({
-      option: { metric_name: 'a_saved_metric', expression: 'COUNT(*)' },
+const renderMetricDefinitionOption = props => {
+  waitFor(() => {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx:25]: This `waitFor` is not necessary as you are already waiting for the elements in your tests.

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx [hunk 2]
```diff
@@ -16,38 +16,38 @@
...
+      useRouter: true,
     });
-    expect(wrapper.find(StyledMetricOption)).toExist();
   });
+};
 
-  it('renders a StyledColumnOption given a column', () => {
-    const wrapper = setup({ option: { column_name: 'a_column' } });
-    expect(wrapper.find(StyledColumnOption)).toExist();
+test('MetricDefinitionOption - renders a given saved metric and display SQL expression popover when hovered', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx:33]:
```suggestion
test('renders a given saved metric and display SQL expression popover when hovered', async () => {
```

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx [hunk 3]
```diff
@@ -16,38 +16,38 @@
...
-    const wrapper = setup({ option: { aggregate_name: 'an_aggregate' } });
-    expect(wrapper.find(AggregateOption)).toExist();
-  });
+  // Grab calculator icon and mock mouse hovering over it
+  const calculatorIcon = await screen.findByLabelText('calculator');
+  userEvent.hover(calculatorIcon);
+  expect(await screen.findByText('SQL expression')).toBeInTheDocument();
+});
+
+test('MetricDefinitionOption - renders when given a column', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx:45]:
```suggestion
test('renders when given a column', async () => {
```

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx [hunk 4]
```diff
@@ -16,38 +16,38 @@
...
+  renderMetricDefinitionOption({ option: { column_name: 'a_column' } });
+  expect(await screen.findByText('a_column')).toBeInTheDocument();
+});
+
+test('MetricDefinitionOption - renders when given an aggregate metric', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx:50]:
```suggestion
test('renders when given an aggregate metric', async () => {
```

**Review by michael-s-molina — CHANGES_REQUESTED** [review]
Thank you for the PR @hainenber. I left some suggestions. Could you also convert this file to Typescript? It should be pretty straightforward.

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx (hunk 1 above)
Comment (hainenber) [reply superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.jsx:25]: Removed in HEAD :D

**Comment by hainenber** [comment]
> Thank you for the PR @hainenber. I left some suggestions. Could you also convert this file to Typescript? It should be pretty straightforward.

Converted to `*.tsx` but not sure if I need to define a proper prop type since it's for testing only 🤔

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import MetricDefinitionOption from 'src/explore/components/controls/MetricControl/MetricDefinitionOption';
+import userEvent from '@testing-library/user-event';
+
+// @ts-ignore-next-line
+const renderMetricDefinitionOption = props => {
+  render(<MetricDefinitionOption {...props} />, {
+    useRedux: true,
+    useRouter: true,
+  });
+};
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx:30]: You can define the type in the test. Unless you want to also convert `MetricDefinitionOption` to Typescript, which would then define the props type and you would import it here.

```suggestion
type MetricDefinitionOptionProps = {
  option: {
    metric_name?: string;
    expression?: string;
    column_name?: string;
    aggregate_name?: string;
  };
};

const renderMetricDefinitionOption = (props: MetricDefinitionOptionProps) => {
  render(<MetricDefinitionOption {...props} />, {
    useRedux: true,
    useRouter: true,
  });
};
```

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+test('renders a given saved metric and display SQL expression popover when hovered', async () => {
+  // @ts-ignore-next-line
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx:33]:
```suggestion
```

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  expect(await screen.findByText('a_saved_metric')).toBeInTheDocument();
+
+  // Grab calculator icon and mock mouse hovering over it
+  const calculatorIcon = await screen.findByLabelText('calculator');
+  userEvent.hover(calculatorIcon);
+  expect(await screen.findByText('SQL expression')).toBeInTheDocument();
+});
+
+test('renders when given a column', async () => {
+  // @ts-ignore-next-line
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx:46]:
```suggestion
```

File: superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  renderMetricDefinitionOption({ option: { column_name: 'a_column' } });
+  expect(await screen.findByText('a_column')).toBeInTheDocument();
+});
+
+test('renders when given an aggregate metric', async () => {
+  // @ts-ignore-next-line
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/MetricControl/MetricDefinitionOption.test.tsx:52]:
```suggestion
```

**Comment by hainenber** [comment]
Thanks Michael for your lead, it's very helpful tip! 

I committed all of your proposed change :D

**Comment by hainenber** [comment]
Ouch, got a failed test in CI but looks like to be flaky one. I ran it successfully in my local machine

image

**Comment by rusackas** [comment]
Re-running it... let's see what happens :)

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the PR and for addressing the comments @hainenber!

**Pull request merged**